src/coreComponents/constitutive/fluid/multifluid/compositional/functions/CompositionalProperties.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file CompositionalProperties.hpp
+ */
+
+#include "CompositionalProperties.hpp"
+
+namespace geos
+{
+
+namespace constitutive
+{
+
+GEOS_HOST_DEVICE
+void CompositionalProperties::computeMolarDensity( integer const numComps,
+                                                   real64 const pressure,
+                                                   real64 const temperature,
+                                                   arraySlice1d< real64 const > const & composition,
+                                                   arraySlice1d< real64 const > const & volumeShift,
+                                                   real64 const compressibilityFactor,
+                                                   real64 & molarDensity )
+{
+
+  real64 vEos = gasConstant * temperature * compressibilityFactor / pressure;
+  real64 vCorrected = vEos;
+
+  for( integer ic = 0; ic < numComps; ++ic )
+  {
+    vCorrected = vCorrected + composition[ic] * ( volumeShift[ic] * temperature + volumeShift[ic] );
```

Comment:
I am puzzled by unit homogeneity ...

Reply:
Nice catch. I think the idea in PVT package is to have 2 values with the second one dependent on temperature. But I'm not quite sure this was coded correctly. I think I will remove this and stick to the formula that is better known without the temperature dependency.